system/src/system_sleep.cpp

```diff
@@ -83,9 +83,16 @@ static int system_sleep_network_resume(network_interface_index index) {
 }
 
 int system_sleep_ext(const hal_sleep_config_t* config, hal_wakeup_source_base_t** reason, void* reserved) {
-    SYSTEM_THREAD_CONTEXT_SYNC(system_sleep_ext(config, reason, reserved));
-
     LOG(TRACE, "Entering system_sleep_ext()");
+    // Cancel current connection attempt to unblock the system thread
+    if (network_connecting(NETWORK_INTERFACE_ALL, 0, NULL)) {
```

Comment:
Let's do this only for Gen 2 platforms where the system loop will be blocked on the networking operation. Gen 3 platforms should not be affected by this.
```c++
#if HAL_PLATFORM_GEN == 2
#endif // HAL_PLATFORM_GEN == 2
```